haystack/components/websearch/serpex.py

```diff
@@ -0,0 +1,207 @@
+        :param engine: Override the default search engine. If None, uses the engine from initialization.
+        :param num_results: Override the default number of results. If None, uses num_results from initialization.
+        :param time_range: Time range filter for results. Options: "all", "day", "week", "month", "year".
+                          Defaults to None (all time).
+        :returns: Dictionary containing a list of Document objects with search results.
+        """
+        documents: list[Document] = []
+
+        try:
+            # Prepare request parameters
+            params: dict[str, Any] = {
+                "q": query,
+                "engine": engine or self.engine,
+                "num": num_results or self.num_results,
+                "category": "web",
+            }
+
+            if time_range:
+                params["time_range"] = time_range
+
+            headers = {"Authorization": f"Bearer {self.api_key.resolve_value()}", "Content-Type": "application/json"}
+
+            # Make API request
+            response = self._make_request("https://api.serpex.dev/api/search", headers, params)
+            data = response.json()
+
+            # Parse search results (API returns 'results' not 'organic_results')
+            if "results" in data and isinstance(data["results"], list):
+                for result in data["results"]:
+                    # Extract result data
+                    title = result.get("title", "")
+                    url = result.get("url", "")  # API uses 'url' not 'link'
+                    snippet = result.get("snippet", "")
+                    position = result.get("position", 0)
+
+                    # Create Document object
+                    doc = Document(
+                        content=snippet,
+                        meta={
+                            "title": title,
+                            "url": url,
+                            "position": position,
+                            "query": query,
+                            "engine": engine or self.engine,
+                        },
+                    )
+                    documents.append(doc)
+
+                logger.info(
+                    "Successfully fetched {count} search results for query: {query}",
+                    count=len(documents),
+                    query=query,
+                )
+            else:
+                logger.warning(
+                    "No results found in SERPEX API response for query: {query}",
+                    query=query,
+                )
+
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                "HTTP error occurred while fetching SERPEX results: {status} - {detail}",
+                status=e.response.status_code,
+                detail=str(e),
+            )
+            raise
+        except httpx.RequestError as e:
+            logger.error("Request error occurred while fetching SERPEX results: {error}", error=str(e))
```

Comment:
The error will be auto formatted into a string without needing to call `str(e)`. It's nice to attach the actual error object so downstream applications can inspect the error object attached to logs. 
```suggestion
            logger.error("Request error occurred while fetching SERPEX results: {error}", error=e)
```